scripts/prepare_heat_demand.py

```diff
@@ -0,0 +1,305 @@
+                temperature=temperature,
+                shlp_type="MFH",
+                building_class=2,
+                wind_class=0,
+                annual_heat_demand=yearly_demands["mfh" + "_" + type],
+                name="MFH",
+                ww_incl=True,
+            ).get_bdew_profile()
+
+            # Calculate industry, trade, service (ghd: Gewerbe, Handel, Dienstleistung)
+            # heat load
+            demand["ghd" + "_" + type] = bdew.HeatBuilding(
+                demand.index,
+                holidays=holidays,
+                temperature=temperature,
+                shlp_type="ghd",
+                wind_class=0,
+                annual_heat_demand=yearly_demands["ghd" + "_" + type],
+                name="ghd",
+                ww_incl=True,
+            ).get_bdew_profile()
+
+        # Calculate total heat load in year
+        heat_load_year = pd.DataFrame(
+            data={"Heat_load_" + str(region) + "_" + str(year): demand.sum(axis=1)}
+        )
+
+        # Stack time series with total heat load in year
+        heat_load_year_stacked = dp.stack_timeseries(heat_load_year)
+
+        # Add index, region and scenario information time series with total heat load
+        # in year
+        heat_load_year_stacked["id_ts"] = index
+        heat_load_year_stacked["region"] = region
+        heat_load_year_stacked["scenario"] = scenario
+        heat_load_year_stacked["var_unit"] = sc_demand_unit
+
+        # Rearrange stacked time series
+        heat_load_year_stacked = dp.format_header(
+            df=heat_load_year_stacked, header=ts_template.columns, index_name="id_ts"
+        )
+
+        # Append stacked heat load of year to stacked time series with total heat loads
+        heat_load = pd.concat([heat_load, heat_load_year_stacked], ignore_index=True)
+
+    return heat_load
+
+
+if __name__ == "__main__":
+    # path_this_file = os.path.realpath(__file__)
+    # raw_data = os.path.abspath(
+    #     os.path.join(path_this_file, os.pardir, os.pardir, "raw")
+    # )
+    # schema = os.path.abspath(
+    #     os.path.join(path_this_file, os.pardir, os.pardir, "oemof_b3", "schema")
+    # )
+    #
+    # in_path1 = os.path.join(raw_data)  # path to raw data
+    # in_path2 = os.path.join(
+    #     raw_data, "distribution_households.csv"
+    # )  # path to household distributions data
+    # in_path3 = os.path.join(raw_data, "holidays.csv")  # path to holidays
+    # in_path4 = os.path.join(raw_data, "scalars.csv")  # path to b3 schema scalars.csv
+    # in_path5 = os.path.join(schema, "timeseries.csv")
+    # out_path = os.path.join(raw_data, "heat_load.csv")
+
+    in_path1 = sys.argv[1]  # path to raw data
+    in_path2 = sys.argv[2]  # path to household distributions data
+    in_path3 = sys.argv[3]  # path to holidays
+    in_path4 = sys.argv[4]  # path to b3 scalars.csv
+    in_path5 = sys.argv[5]  # path to template of times series format
+    out_path = sys.argv[6]
+
+    REGION = "BB"
```

Comment:
You need to loop over region, year and carrier - could put the loop for all three here, outside of the function.

Reply:
Is carrier suppose to have decentral or central in it? So far I assumed that the carrier is always `heat` and the distinction in central and decentral is in the variable `name`.

Reply:
The question has been resolved with your [comment above](https://github.com/rl-institut/oemof-B3/pull/76#discussion_r749480062).

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/76/commits/ae095e268fdc66e2277c389abf1a33e7abe29270.